concepts/recursion/about.md

```diff
@@ -0,0 +1,102 @@
+# About
+
+Recursive functions are functions that call themselves.
+
+A recursive function needs to have at least one _base case_ and at least one _recursive case_.
+
+A _base case_ returns a value without calling the function again.
+A _recursive case_ calls the function again, modifying the input so that it will at some point match the base case.
+
+In Elm, recursive functions are typically defined using pattern matching with `case` expressions.
+
+```elm
+count : List a -> Int
+count list =
+    case list of
+        [] ->
+            0  -- base case
+
+        _ :: tail ->
+            1 + count tail  -- recursive case
+```
+
+A recursive function can have many base cases and/or many recursive cases.
+For example [the Fibonacci sequence][fibonacci] is a recursive sequence with two base cases:
+
+```elm
+fibonacci : Int -> Int
+fibonacci n =
+    case n of
+        0 ->
+            0
+
+        1 ->
+            1
+
+        _ ->
+            fibonacci (n - 1) + fibonacci (n - 2)
+```
+
+Counting the number of occurrences of some given value `x` in a list has two recursive cases:
+
+```elm
+countOccurrences : List a -> a -> Int
+countOccurrences list x =
+    case list of
+        [] ->
+            0
+
+        head :: tail ->
+            if head == x then
+                1 + countOccurrences tail x
+            else
+                countOccurrences tail x
+```
+
+## Loops through recursion
+
+Due to immutability, loops in Elm are written differently from imperative languages.
+For example, loops commonly look like:
+
+```
```

Comment:
Maybe mark this as c or something?

Reply:
It's not C, because of `array.length`. I'll mention it's pseudocode